components/permissions_gates/channel_permission_gate/channel_permission_gate.jsx

```diff
@@ -7,7 +7,9 @@ import React from 'react';
 export default class ChannelPermissionGate extends React.Component {
     shouldComponentUpdate(nextProps) {
         return this.props.invert !== nextProps.invert ||
-               this.props.hasPermission !== nextProps.hasPermission;
+               this.props.hasPermission !== nextProps.hasPermission ||
+               this.props.channelId !== nextProps.channelId ||
+               this.props.teamId !== nextProps.teamId;
```

Comment:
This was causing the children components to have stale props.

Reply:
I don't know why, the component must be rerendered only if the invert or hasPermission has changed, it is checked in every loop of the redux based on the component properties. I may be failing, but I don't understand why.

Reply:
In the channel drop-down if you change channels and pass it to the permissions gate the children were still using the old channel id.

<img width="223" alt="screen shot 2018-03-26 at 9 30 11 am" src="https://user-images.githubusercontent.com/1149597/37909215-5314faa0-30d8-11e8-9547-dc392694cd2f.png">

Reply:
Yes, you are completely right, but what I think we need to check is the "children", not the other props.

Reply:
Could you explain? I'm not seeing the advantage to that.

Reply:
Of course. Now it only rerenders when your permissions to view the content changes (invert + hasPermission), if you add the children, you cover all the "render" method properties. Imagine that you want to show a text depending on your permissions. The children will never change, so, only rerender on change of permissions (invert + hasPermissions), and what is more important, imagine that you have a permission gate that contains a block which have a property which is not channelId or teamId, if that property changes and none of the other, it will not be reflected in the permission gate rendering, because the permission gate is not re-rendered.

Reply:
I just did a test and `this.props.children !== nextProps.children` re-renders 68x (every time) vs 14x with `this.props.channelId !== nextProps.channelId || this.props.teamId !== nextProps.teamId`. Thoughts?

Reply:
I think we may just need to remove the `shouldComponentUpdate` method from the `...Gate` components.

Reply:
Yes, I reach the same conclusion.